contrib/endpoints/src/api_manager/request_handler_test.cc

```diff
@@ -0,0 +1,460 @@
+    return true;
+  }
+
+  ::google::api_manager::protocol::Protocol GetFrontendProtocol() {
+    return ::google::api_manager::protocol::Protocol::HTTP;
+  }
+
+  ::google::api_manager::protocol::Protocol GetBackendProtocol() {
+    return ::google::api_manager::protocol::Protocol::HTTPS;
+  }
+
+  void SetAuthToken(const std::string &auth_token) {}
+
+  utils::Status AddHeaderToBackend(const std::string &key,
+                                   const std::string &value) {
+    return utils::Status::OK;
+  }
+
+ private:
+  std::unordered_map<std::string, std::string> data_;
+};
+
+class ResponseMock : public Response {
+ public:
+  virtual ~ResponseMock() {}
+  utils::Status GetResponseStatus() { return utils::Status::OK; }
+  std::size_t GetRequestSize() { return 0; }
+  std::size_t GetResponseSize() { return 0; }
+  utils::Status GetLatencyInfo(service_control::LatencyInfo *info) {
+    return utils::Status::OK;
+  }
+};
+
+class RequestHandlerTest : public ::testing::Test {
+ protected:
+  RequestHandlerTest() : callback_run_count_(0), raw_env_(nullptr) {}
+
+  void SetUp() {
+    callback_run_count_ = 0;
+
+    env_.reset(new ::testing::NiceMock<MockApiManagerEnvironment>());
+    raw_env_ = env_.get();
+
+    api_manager_ = std::make_shared<ApiManagerImpl>(
+        std::move(env_), "", kServerConfigWithServiceNameConfigId);
+    EXPECT_TRUE(api_manager_);
+
+    request_handler_ = api_manager_->CreateRequestHandler(
+        std::unique_ptr<Request>(new RequestMock({{"method", "GET"},
+                                                  {"ip", "127.0.0.1"},
+                                                  {"host", "localhost"},
+                                                  {"path", "/echo"}})));
+  }
+
+ protected:
+  int callback_run_count_;
+  std::shared_ptr<ApiManagerImpl> api_manager_;
+  std::unique_ptr<MockApiManagerEnvironment> env_;
+  MockApiManagerEnvironment *raw_env_;
+  std::unique_ptr<RequestHandlerInterface> request_handler_;
+
+ private:
+  ApiManagerFactory factory_;
+};
+
+TEST_F(RequestHandlerTest, PendingCheckApiManagerInitSucceeded) {
+  EXPECT_CALL(*raw_env_, DoRunHTTPRequest(_))
+      .WillOnce(Invoke([this](HTTPRequest *req) {
+        EXPECT_EQ(
+            "https://servicemanagement.googleapis.com/v1/services/"
+            "bookstore.test.appspot.com/configs/2017-05-01r0",
+            req->url());
+        req->OnComplete(Status::OK, {}, std::move(kServiceConfig1));
+      }))
+      .WillOnce(Invoke([this](HTTPRequest *req) {
+        EXPECT_EQ("http://localhost/computeMetadata/v1/?recursive=true",
+                  req->url());
+        req->OnComplete(Status::OK, {}, "{}");
+
+      }));
+
+  EXPECT_EQ("UNAVAILABLE: Not initialized yet",
+            api_manager_->ConfigLoadingStatus().ToString());
+  EXPECT_EQ("bookstore.test.appspot.com", api_manager_->service_name());
+  EXPECT_EQ("", api_manager_->service("2017-05-01r0").id());
+
+  request_handler_->Check([this](utils::Status status) {
```

Comment:
please add comment saying:
// api-key is not required and api-key is not provided, so 
// not need to make remote call for Check().